test/Microsoft.ML.Tests/Transformers/CategoricalHashTests.cs

```diff
@@ -74,11 +75,17 @@ public void CategoricalHashStatic()
             var invalidData = ComponentCreation.CreateDataView(Env, wrongCollection);
             var est = data.MakeNewEstimator().
                   Append(row => (
-                  A: row.ScalarString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashScalarOutputKind.Ind),
-                  B: row.VectorString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashVectorOutputKind.Ind),
-                  C: row.VectorString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashVectorOutputKind.Bag),
-                  D: row.ScalarString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashScalarOutputKind.Bin),
-                  E: row.VectorString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashVectorOutputKind.Bin)
+                      row.ScalarString,
+                      row.VectorString,
+                      // Create a VarVector column
+                      VarVectorString: row.ScalarString.TokenizeText())).
+                  Append(row => (
+                      A: row.ScalarString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashScalarOutputKind.Ind),
+                      B: row.VectorString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashVectorOutputKind.Ind),
+                      C: row.VectorString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashVectorOutputKind.Bag),
+                      D: row.ScalarString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashScalarOutputKind.Bin),
+                      E: row.VectorString.OneHotHashEncoding(outputKind: CategoricalHashStaticExtensions.OneHotHashVectorOutputKind.Bin),
+                      F: row.VarVectorString.OneHotHashEncoding()
```

Comment:
Do you need to test any variations in input parameters?

Reply:
The default is Bag which works and makes sense. For Ind, the output makes sense for Vector but I don't think it makes sense for VarVector. It would be variable vectors of indicators, which can't be fed into a learner and I can't think of another transform to use it.

---
In reply to: [236895571](https://github.com/dotnet/machinelearning/pull/1745#discussion_r236895571) [](ancestors = 236895571)